core/amber/src/main/scala/edu/uci/ics/texera/workflow/operators/visualization/IcicleChart/IcicleChartOpDesc.scala

```diff
@@ -0,0 +1,113 @@
+package edu.uci.ics.texera.workflow.operators.visualization.IcicleChart
+
+import com.fasterxml.jackson.annotation.{JsonProperty, JsonPropertyDescription}
+import com.kjetland.jackson.jsonSchema.annotations.{JsonSchemaInject, JsonSchemaTitle}
+import edu.uci.ics.texera.workflow.common.metadata.annotations.AutofillAttributeName
+import edu.uci.ics.texera.workflow.common.metadata.{OperatorGroupConstants, OperatorInfo}
+import edu.uci.ics.texera.workflow.common.operators.PythonOperatorDescriptor
+import edu.uci.ics.amber.engine.common.workflow.{InputPort, OutputPort}
+import edu.uci.ics.texera.workflow.common.tuple.schema.{Attribute, AttributeType, Schema}
+import edu.uci.ics.texera.workflow.operators.visualization.{
+  VisualizationConstants,
+  VisualizationOperator
+}
+import edu.uci.ics.texera.workflow.operators.visualization.hierarchychart.HierarchySection
+
+// type constraint: value can only be numeric
+@JsonSchemaInject(json = """
+{
+  "attributeTypeRules": {
+    "value": {
+      "enum": ["integer", "long", "double"]
+    }
+  }
+}
+""")
+class IcicleChartOpDesc extends VisualizationOperator with PythonOperatorDescriptor {
+  @JsonProperty(required = true)
+  @JsonSchemaTitle("Hierarchy Path")
+  @JsonPropertyDescription(
+    "hierarchy of attributes from a root (higher-level category) to leaves (lower-level category)"
+  )
+  var hierarchy: List[HierarchySection] = List()
+
+  @JsonProperty(value = "value", required = true)
+  @JsonSchemaTitle("Value Column")
+  @JsonPropertyDescription("the value associated with the size of each sector in the chart")
+  @AutofillAttributeName
+  var value: String = ""
+
+  override def getOutputSchema(schemas: Array[Schema]): Schema = {
+    Schema.builder().add(new Attribute("html-content", AttributeType.STRING)).build()
+  }
+
+  override def operatorInfo: OperatorInfo =
+    OperatorInfo(
+      "Icicle Chart",
+      "Visualize hierarchical data from root to leaves",
+      OperatorGroupConstants.VISUALIZATION_GROUP,
+      inputPorts = List(InputPort()),
+      outputPorts = List(OutputPort())
+    )
+
+  private def getIcicleAttributesInPython: String =
+    hierarchy.map(_.attributeName).mkString("'", "','", "'")
+
+  def manipulateTable(): String = {
+    assert(value.nonEmpty)
```

Comment:
remove this assert. We usually don't have such assertion here. We only used it in unit tests. As the JsonProperty already enforced it as required, then you don't need to do the check here